File: src/main/core/io/github/kloping/Mcore/java/ListenerHosts/BaseMessageListener.java
package io.github.kloping.Mcore.java.ListenerHosts;

import io.github.kloping.Mcore.java.BotStarter;
import io.github.kloping.Mcore.java.utils.MessageUtils;
import io.github.kloping.MySpringTool.Starter;
import io.github.kloping.MySpringTool.StarterApplication;
import kotlin.coroutines.CoroutineContext;
import net.mamoe.mirai.contact.Contact;
import net.mamoe.mirai.event.EventHandler;
import net.mamoe.mirai.event.SimpleListenerHost;
import net.mamoe.mirai.event.events.BotOfflineEvent;
import net.mamoe.mirai.event.events.FriendMessageEvent;
import net.mamoe.mirai.event.events.GroupMessageEvent;
import net.mamoe.mirai.event.events.StrangerMessageEvent;
import net.mamoe.mirai.message.data.Message;

/**
 * @author github-kloping
 */
public class BaseMessageListener extends SimpleListenerHost {
    public BaseMessageListener() {
        super();
    }

    public BaseMessageListener(CoroutineContext coroutineContext) {
        super(coroutineContext);
    }

    @Override
    public void handleException(CoroutineContext context, Throwable exception) {
        super.handleException(context, exception);
    }

    /**
     * 收到群消息
     *
     * @param event
     */
    @EventHandler
    public void handGroupMessage(GroupMessageEvent event) {
        long q = event.getSender().getId();
        Contact contact = event.getSubject();
        Message message = event.getMessage();
        String text = MessageUtils.toText(event.getMessage()).trim();
        StarterApplication.executeMethod(q, text, q, contact, message);
    }

    /**
     * 收到朋友消息
     *
     * @param event
     */
    @EventHandler
    public void handFriendMessage(FriendMessageEvent event) {
        long q = event.getSender().getId();
        Contact contact = event.getSubject();
        Message message = event.getMessage();
        String text = MessageUtils.toText(event.getMessage()).trim();
        StarterApplication.executeMethod(q, text, q, contact, message);
    }

    /**
     * 陌生人消息
     *
     * @param event
     */
    @EventHandler
    public void handStrangerMessage(StrangerMessageEvent event) {

    }

    /**
     * 机器人掉线
     *
     * @param event
     */
    @EventHandler
    public void onLost(BotOfflineEvent event) {
    }
}

File: src/main/core/io/github/kloping/Mcore/kotlin/ListenerHosts/BaseMessageListener.kt
package io.github.kloping.Mcore.kotlin.ListenerHosts

import io.github.kloping.Mcore.kotlin.BotStarter
import io.github.kloping.Mcore.kotlin.utils.MessageUtils
import io.github.kloping.MySpringTool.Starter
import io.github.kloping.MySpringTool.StarterApplication
import net.mamoe.mirai.contact.Contact
import net.mamoe.mirai.event.EventHandler
import net.mamoe.mirai.event.SimpleListenerHost
import net.mamoe.mirai.event.events.BotOfflineEvent
import net.mamoe.mirai.event.events.FriendMessageEvent
import net.mamoe.mirai.event.events.GroupMessageEvent
import net.mamoe.mirai.event.events.StrangerMessageEvent
import net.mamoe.mirai.message.data.Message
import kotlin.coroutines.CoroutineContext

/**
 * @author github-kloping
 */
class BaseMessageListener : SimpleListenerHost {
    constructor() : super() {}
    constructor(coroutineContext: CoroutineContext?) : super(coroutineContext!!) {}

    override fun handleException(context: CoroutineContext, exception: Throwable) {
        super.handleException(context, exception)
    }

    /**
     * 收到群消息
     *
     * @param event
     */
    @EventHandler
    fun handGroupMessage(event: GroupMessageEvent) {
        val q = event.sender.id
        val contact: Contact = event.subject
        val message: Message = event.message
        val text = MessageUtils.toText(event.message).trim { it <= ' ' }
        StarterApplication.executeMethod(q, text, q, contact, message)
    }

    /**
     * 收到朋友消息
     *
     * @param event
     */
    @EventHandler
    fun handFriendMessage(event: FriendMessageEvent) {
        val q = event.sender.id
        val contact: Contact = event.subject
        val message: Message = event.message
        val text = MessageUtils.toText(event.message).trim { it <= ' ' }
        StarterApplication.executeMethod(q, text, q, contact, message)
    }

    /**
     * 陌生人消息
     *
     * @param event
     */
    @EventHandler
    fun handStrangerMessage(event: StrangerMessageEvent?) {
    }

    /**
     * 机器人掉线
     *
     * @param event
     */
    @EventHandler
    fun onLost(event: BotOfflineEvent?) {
        Starter.reLoadConfigurationFile(BotStarter.confFile)
        var k = Starter.getContextValue<Boolean>(Boolean::class.javaPrimitiveType, "ReLogin")
        k = k ?: BotStarter.autoReLogin
        BotStarter.bot!!.configuration.autoReconnectOnForceOffline = k
    }
}